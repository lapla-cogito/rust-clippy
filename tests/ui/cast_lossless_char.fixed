#![warn(clippy::cast_lossless)]
#![allow(clippy::char_lit_as_u8)]

type I32 = i32;
type U32 = u32;

fn main() {
    let _ = u32::from('a');
    //~^ cast_lossless
    let _ = 'a' as i32;
    let _ = u64::from('a');
    //~^ cast_lossless
    let _ = 'a' as i64;
    let _ = U32::from('a');
    //~^ cast_lossless
    let _ = 'a' as I32;

    let _ = 'a' as u8;
    let _ = 'a' as i8;
}
