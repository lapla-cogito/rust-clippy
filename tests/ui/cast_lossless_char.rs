#![warn(clippy::cast_lossless)]
#![allow(clippy::char_lit_as_u8)]

type I32 = i32;
type U32 = u32;

fn main() {
    let _ = 'a' as u32;
    //~^ cast_lossless
    let _ = 'a' as i32;
    let _ = 'a' as u64;
    //~^ cast_lossless
    let _ = 'a' as i64;
    let _ = 'a' as U32;
    //~^ cast_lossless
    let _ = 'a' as I32;

    let _ = 'a' as u8;
    let _ = 'a' as i8;
}
