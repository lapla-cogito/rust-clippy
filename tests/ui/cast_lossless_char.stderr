error: casts from `char` to `u32` can be expressed infallibly using `From`
  --> tests/ui/cast_lossless_char.rs:8:13
   |
LL |     let _ = 'a' as u32;
   |             ^^^^^^^^^^
   |
   = help: an `as` cast can become silently lossy if the types change in the future
   = note: `-D clippy::cast-lossless` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::cast_lossless)]`
help: use `u32::from` instead
   |
LL -     let _ = 'a' as u32;
LL +     let _ = u32::from('a');
   |

error: casts from `char` to `u64` can be expressed infallibly using `From`
  --> tests/ui/cast_lossless_char.rs:11:13
   |
LL |     let _ = 'a' as u64;
   |             ^^^^^^^^^^
   |
   = help: an `as` cast can become silently lossy if the types change in the future
help: use `u64::from` instead
   |
LL -     let _ = 'a' as u64;
LL +     let _ = u64::from('a');
   |

error: casts from `char` to `u32` can be expressed infallibly using `From`
  --> tests/ui/cast_lossless_char.rs:14:13
   |
LL |     let _ = 'a' as U32;
   |             ^^^^^^^^^^
   |
   = help: an `as` cast can become silently lossy if the types change in the future
help: use `U32::from` instead
   |
LL -     let _ = 'a' as U32;
LL +     let _ = U32::from('a');
   |

error: aborting due to 3 previous errors

